momento-provider/src/test/java/com/google/code/ssm/providers/momento/TestPOC.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.io.Serializable;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(loader= AnnotationConfigContextLoader.class)
+public class TestPOC {
+    @Configuration
+    public static class ContextConfiguration extends AbstractSSMConfiguration {
+        @Bean
+        @Override
+        public CacheFactory defaultMemcachedClient() {
+            final MomentoConfiguration conf = new MomentoConfiguration();
+            conf.setConsistentHashing(true);
+            conf.setDefaultTtl(300);
+            // IMPORTANT: This presumes you have already created a cache in Momento named 'example-cache'
+            conf.setDefaultCacheName("example-cache");
+            conf.setMomentoAuthToken("<YOUR AUTH TOKEN GOES HERE>");
+            final CacheFactory cf = new CacheFactory();
+            cf.setCacheClientFactory(new MomentoClientFactoryImpl());
+            // Doesn't actually do anything
+            cf.setAddressProvider(new DefaultAddressProvider("127.0.0.1:11211"));
+            cf.setConfiguration(conf);
+            return cf;
+        }
+    }
+    @Autowired
+    private CacheFactory cacheFactory;
+
+    @Data
+    @JsonSerialize
+    public static class TestObject implements Serializable {
+        int id;
+        String name;
+    }
+
+
+    @ReadThroughSingleCache(namespace = "TestString", expiration = 3600)
+    @SneakyThrows
+    public String getSimpleObject(@ParameterValueKeyProvider String complexObjectPk) {
+        Cache cache = cacheFactory.getObject();
+        if (cache != null) {
+            String result = cache.get(complexObjectPk, SerializationType.PROVIDER);
+            if (result != null) {
+                return result;
+            }
+        }
+        // Otherwise, retrieve object from source location, e.g. DynamoDB, S3
+        // return getObjectFromSource(complexObjectPk);
+        return null;
+    }
+
+    @ReadThroughSingleCache(namespace = "TestObject", expiration = 3600)
+    @SneakyThrows
+    public TestObject getComplexObject(@ParameterValueKeyProvider String objectKey) {
+        Cache cache = cacheFactory.getObject();
+        ObjectMapper mapper = new ObjectMapper();
+        if (cache != null) {
+            TestObject obj = mapper.readValue((String) cache.get(objectKey, SerializationType.PROVIDER), TestObject.class);
+            if (obj != null) {
+                return obj;
+            }
+        }
+        // Otherwise, retrieve object from source location, e.g. DynamoDB, S3
+        // return getObjectFromSource(objectKey);
+        return null;
+    }
+
+
+    @Test
+    @SneakyThrows
+    public void testPocSimple() {
+        String test = "fooBar";
+        cacheFactory.getObject().set("test-key", 500, test, SerializationType.PROVIDER);
+        String result = getSimpleObject("test-key");
```

Comment:
This shouldn't work right? What does `namespace` do above?

Reply:
This works, the unit tests (when the `@Ignore` annotation is removed) shows that it works. `cacheFactory.getObject()` actually returns the underlying `Cache` object itself.

As for namespace... no idea what that actually does